pkg/cmd/experimental/syncgroups/ad/ldapinterface.go

```diff
@@ -0,0 +1,114 @@
+package ad
+
+import (
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/go-ldap/ldap"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+)
+
+// NewLDAPInterface builds a new LDAPInterface using a schema-appropriate config
+func NewLDAPInterface(clientConfig ldaputil.LDAPClientConfig,
+	userQuery ldaputil.LDAPQueryOnAttribute,
+	groupMembershipAttributes []string,
+	userNameAttributes []string) LDAPInterface {
+	return LDAPInterface{
+		clientConfig:              clientConfig,
+		userQuery:                 userQuery,
+		userNameAttributes:        userNameAttributes,
+		groupMembershipAttributes: groupMembershipAttributes,
+		cachedUsers:               map[string]*ldap.Entry{},
+		ldapGroupToLDAPMembers:    map[string][]*ldap.Entry{},
+	}
+}
+
+var _ interfaces.LDAPMemberExtractor = &LDAPInterface{}
```

Comment:
> move to test?

I like having the intent very clear in the code for people looking at the struct.  I'll move the other to match.